baseapp/baseapp_payments/serializers.py

```diff
@@ -12,29 +12,94 @@
 logger = logging.getLogger(__name__)
 
 
+STRIPE_ACTIVE_SUBSCRIPTION_STATUSES = {"active", "trialing", "incomplete", "past_due"}
+
+
 class StripeSubscriptionSerializer(serializers.Serializer):
-    remote_customer_id = serializers.CharField()
-    price_id = serializers.CharField()
-    remote_subscription_id = serializers.ReadOnlyField()
+    remote_customer_id = serializers.CharField(help_text="Stripe customer ID")
+    price_id = serializers.CharField(help_text="Stripe price ID")
+    allow_incomplete = serializers.BooleanField(
+        default=False,
+    )
+    payment_method_id = serializers.CharField(
+        required=False,
+    )
+    billing_details = serializers.DictField(
+        required=False,
+    )
+
+    remote_subscription_id = serializers.CharField(read_only=True)
+    client_secret = serializers.CharField(read_only=True)
+    invoice_id = serializers.CharField(read_only=True)
+    latest_invoice = serializers.DictField(read_only=True)
 
     def validate(self, data):
+        customer_id = data["remote_customer_id"]
```

Comment:
I might have missed it, but it looks like we're not validating that the customer is the authenticated user, can you confirm if we are, and fix if we're not?